### PR TITLE
task done

### PR DESCRIPTION
**Pull request by yutsevich**

### CODE REVIEW

**Review by andriylagun — APPROVED** [review]
Good job!

File: src/main/java/com/internet/shop/controllers/AddProductController.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/product/add.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String priceStr = req.getParameter("price");
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/AddProductController.java]:
```suggestion
        String price = req.getParameter("price");
```

File: src/main/java/com/internet/shop/controllers/AddProductController.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+        if (name != null && priceStr != null && !name.equals("") && !priceStr.equals("")) {
+            productService.create(new Product(name,Long.parseLong(priceStr)));
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/AddProductController.java]:
```suggestion
            productService.create(new Product(name, Long.parseLong(priceStr)));
```

File: src/main/java/com/internet/shop/controllers/DeleteUserController.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+public class DeleteUserController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String userId = req.getParameter("id");
+        Long id = Long.parseLong(userId);
+        userService.deleteById(id);
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/DeleteUserController.java:24]: давай окрім юзера також видаляти його корзину

File: src/main/webapp/web.xml [hunk 4]
```diff
@@ -2,17 +2,96 @@
...
+    </servlet-mapping>
+    <!--inject data-->
+    <servlet>
+        <servlet-name>inject data</servlet-name>
+        <servlet-class>com.internet.shop.controllers.InjectDataController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>inject data</servlet-name>
+        <url-pattern>/injectData</url-pattern>
```
Comment (Sofasmile) [inline src/main/webapp/web.xml]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/webapp/web.xml [hunk 5]
```diff
@@ -2,17 +2,96 @@
...
+    </servlet-mapping>
+    <!--Add product to cart-->
+    <servlet>
+        <servlet-name>Add product to cart</servlet-name>
+        <servlet-class>com.internet.shop.controllers.AddProductToShoppingCartController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>Add product to cart</servlet-name>
+        <url-pattern>/shoppingCart/product/add</url-pattern>
```
Comment (Sofasmile) [inline src/main/webapp/web.xml]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/webapp/web.xml [hunk 6]
```diff
@@ -2,17 +2,96 @@
...
+    </servlet-mapping>
+    <!--Shopping cart-->
+    <servlet>
+        <servlet-name>Shopping cart</servlet-name>
+        <servlet-class>com.internet.shop.controllers.GetAllProductsFromShoppingCartController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>Shopping cart</servlet-name>
+        <url-pattern>/shoppingCart</url-pattern>
```
Comment (Sofasmile) [inline src/main/webapp/web.xml]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**